[main < 204-expand-getting-started...] Add new page Getting started with Docker (#214)
memgraph/documentation: pages/getting-started/first-steps-with-docker.mdx

{<h3>Verify Docker Version</h3>}

Type the following command:

```
docker --version
```

This should display the version of Docker installed on your machine. For example:

```
Docker version 24.0.6, build ed223bc
```

{<h3>Check Docker Details</h3>}

For more detailed information about the Docker installation, you can use:

```
docker info
```

This command provides a comprehensive overview of your Docker setup, including the number of containers running, paused, and stopped, as well as other system information.

{<h3>Ensure Docker Daemon is Running</h3>}

To check that the Docker daemon (background service) is running, execute:

```
docker system info
```

If Docker is running correctly, this command will return system-wide information. If not, you may receive an error indicating that Docker is not running.

</Steps>

### Troubleshooting

If you encounter any errors or if Docker does not seem to be running:

- Ensure that the Docker Desktop application (for Mac and Windows) is running.
- On Linux, you might need to start the Docker service manually using a command like `sudo service docker start` or `sudo systemctl start docker`, depending on your Linux distribution.

> <Callout type="info">

Reviewer: I think this callout is too huge :/ why not just make it a chapter?

## Managing Docker Service on Windows using PowerShell

On Windows, Docker typically runs as a service via Docker Desktop. While Docker Desktop provides a user-friendly interface for managing Docker, there are instances where you might want to control the Docker service directly from the command line using PowerShell.

Here's how you can manage the Docker service on Windows using PowerShell:

### Open PowerShell as an Administrator

> - Right-click the Start button and select "Windows PowerShell (Admin)" or search for "PowerShell" in the start menu, right-click "Windows PowerShell", and select "Run as administrator".

Reviewer: ```suggestion
Right-click the Start button and select "Windows PowerShell (Admin)" or search for "PowerShell" in the start menu, right-click "Windows PowerShell", and select "Run as administrator".
```

Reviewer: If Windows is using administrator as a lower case administrator, the title should also be administrator, not Administrator

Author: The use Administrator. I've updated the paragraph.

### Start the Docker Service

If Docker isn't running and you wish to start it, use the following command:

```powershell
net start com.docker.service
```

You should receive a message indicating that the Docker service has started successfully.

### Stop the Docker Service

To stop the Docker service, use the following command:

```powershell
net stop com.docker.service
```

You should receive a message indicating that the Docker service has stopped.

### Check the Status of the Docker Service

To verify the status of the Docker service, whether it's running or stopped, use:

```powershell
Get-Service com.docker.service
```

This command will display the status of the Docker service.

</Callout>

## Work with Docker

If you are new to Docker, this guide will help you get a grasp of Docker and
make it easier to accomplish tasks within Memgraph. After installing Docker, all
commands are run from the command-line tool of your preference.

### Download the Memgraph Docker image

> <Callout type="info">

Reviewer: I wouldn't make this a callout, i would make it a regular paragraph ... and kind of incorporate it with the rest of the text... so:

You can think about downloading Memgraph Docker images as obtaining a blueprint for Memgraph, complete with all its core features.  ....

When you pull a Memgraph Docker image, think of it as obtaining the blueprint
for Memgraph, complete with all its core features. This image is your starting
point. From it, you can launch a container, which acts as a live instance of
Memgraph. All your interactions, queries, and transactions with Memgraph occur
within this container. This ensures a consistent and isolated setting for all
your database tasks. From a single image, you can spawn multiple
containers. Each container functions as a distinct Memgraph instance.
</Callout>

Images are downloaded using the `docker pull` command followed by the name of
the Docker image. We encourage you to use Memgraph Platform as it includes
everything you might need while making the best of Memgraph.

To download the latest Memgraph Platform image, run:

```
docker pull memgraph/memgraph-platform
```

All Memgraph Docker images are availble at [Memgraph's Docker
Hub](https://hub.docker.com/u/memgraph). There you will find exact Docker image
tags you can downalod and use to install Memgraph with. 

### Architecture of Docker container running Memgraph

The picture below shows the architecture of the Memgraph Docker ecosystem.

> ![](/pages/getting-started/install-memgraph/docker/docker-architecture.png)

Reviewer: As this is now separate from the main docker installation, maybe here we could list all the images Memgraph has, show this "blueprint" what you get with each image

Also please look at what language i used to described the images and when to use which

### Run a Memgraph Docker image

All images are started using the `docker run` command followed by various flags,
environment variables and configuration options.

The most common flags used while running Memgraph images are:

- publish ports: `-p 3000:3000`
- container name: `--name memgraph`
- set up configuration using environment variables in the case of the
  `memgraph-platform` image, or configuration flags using the `memgraph` or
  `memgraph-mage` image

A `docker run` command can look like this:

```

> docker run -p 7687:7687 [-p host_port:container_port] [--name container_name] [configuration] memgraph/image_name

Reviewer: add --name

```

#### Publish ports

Ports are published in order to allow services outside the container to be able
to connect to the container. When publishing ports, you need to define two ports
separated by a colon. The left side port stands for the **host port** and the
right side port is the **container port**.

The most common ports published while running Memgraph are:

- `-p 7687:7687` - connection to the database instance (the Bolt protocol uses
  this port by default)
- `-p 3000:3000` - connection to the Memgraph Lab application when running
  Memgraph Platform
- `-p 7444:7444` - connection to fetch log files from Memgraph Lab (only in
  version 2.0 and newer)

For example, if you are running two instances using the `memgraph-platform`
image and you want to connect to both instances using the Memgraph Lab
in-browser application. You would run the first instance with:

```
docker run -p 7444:7444 -p 3000:3000 memgraph/memgraph-platform
```

Because port `3000` is now taken, you need to change the left side
port (host ports):

```
docker run -p 7444:7444 -p 3001:3000 memgraph/memgraph-platform

> ```

Reviewer: this you already fixed in the main, but needs fixing here

To connect to the first instance, you should open Memgraph Lab in your browser

> at `localhost:3000`, but the second instance is reachable at `localhost:3001`.

Reviewer: maybe we should also explain that this change in 7444 port is for external services

file from the Docker container, modify it and copy it back into the container.

The logs will be saved to the `mg_log` volume, and the directories can usually be
found in `/var/lib/docker/volumes/`, but you can also view them in the Memgraph
Lab 2.0 (or newer) by publishing the port `7444`.

#### Specify bind mounts

Bind mounts are local directories or files that can be modified by other
processes other than Docker. Any changes made to these directories or files
locally will be reflected inside the Docker container and vice-versa. Also, a
bind mount will overwrite the content of the Docker container.

For example, if I have a `Data` directory on my `C:` disk, and I want to access
it from inside the container at `/usr/lib/memgraph/data`, I would run Docker
with the following `-v` flag.

```
docker run -p 7687:7687 -p 7444:7444 -p 3000:3000  -v "C:/data":/usr/lib/memgraph/data memgraph/memgraph-platform
```

You can use bind mounts to transfer durability files such as snapshot or WAL
files inside the container to restore data, or CSV files you will use to import
data with `CSV LOAD` clause. 

Bind mounts do not handle data permissions which could cause issues with
permissions. 

> ### Stop container

Reviewer: maybe you can add a callout that stopping isn't deleting and link to that last chapter

ID](#how-to-retrieve-a-docker-container-id) or use its [name](#container-name).

Then run the following command if the container ID is `9397623cd87e`.

```console
docker inspect -f '{{range.NetworkSettings.Networks}}{{.IPAddress}}{{end}}' 9397623cd87e
```

### Browse files inside a Docker container

To browse files inside a Docker container, first, you need [to retrieve its
ID](#how-to-retrieve-a-docker-container-id) or use its [name](#container-name).

Then run the following command if the container ID is `9397623cd87e`:

```
docker exec -it 9397623cd87e bash
```

To navigate through the container, use the following commands:

- `ls` - list all the directories and files
- `cd <directory>` - enter a directory
- `cd ..` - leave the directory
- `cat <file>` - list the content of a file

You don't have to write file and directory names in full, once you write enough
letters to have a unique string, press the `<TAB>` key.

### Copy files from and to a Docker container

To copy files from and to the Docker container, first, you need [to retrieve its
ID](#how-to-retrieve-a-docker-container-id) or use its [name](#container-name).

> 1. Place yourself in the local directory where you want to copy the file.

Author: Done.

2. Copy the file from the container to your current directory with the command:

  ```
  docker cp <CONTAINER ID>:<path_in_container> <local_file_name>
  ```

  Be sure to replace the `<CONTAINER ID>` parameter.

  The example below will copy the configuration file to the user's Desktop:

  ```
  C:\Users\Vlasta\Desktop>docker cp bb3de2634afe:/etc/memgraph/memgraph.conf memgraph.conf
  ```

3. Copy the file from your current directory to the container with the command:

  ```
  docker cp  <local_file_name> <CONTAINER ID>:<path_in_container>
  ```

  Be sure to replace the `<CONTAINER ID>` parameter.

  The example below will replace the configuration file with the one from the
  user's Desktop:

  ```
  C:\Users\Vlasta\Desktop>docker cp memgraph.conf bb3de2634afe:/etc/memgraph/memgraph.conf
  ```

> <Callout type="info">

Reviewer: I don't like when whole chapters are info callouts... they are huge + why is this more valuable info than any other on this page?

Author: IMHO callouts have less vaulabe info that is relevant only for some special/edge cases :) I'll convert it to normal paragraph.

## Containers and Data Persistence

When you're working with Docker, it's essential to understand how data is handled in containers. 

> **Your Data is Safe**: Even if you stop a container using the `docker stop` command or if it crashes unexpectedly, the data and changes inside that container are retained. So, when you restart the stopped container, you'll find everything just as you left it.

Reviewer: link to the stopping container section

However, there are a few things to keep in mind:

> - **Deleting a Container**: If you deliberately delete a container using commands like `docker rm`, the changes and data within that container will be lost unless you've saved them in Docker volumes or bind mounts.

Author: We have it in this section.

  
- **Using Volumes**: To ensure long-term data safety and for scenarios where you might delete containers but want to retain their data, Docker provides a feature called volumes. Volumes allow you to store data outside the container, ensuring it remains safe and accessible even if the container is removed.

> In short, while working with containers, always remember: stopping isn't deleting. Your data stays intact when a container is merely stopped. If you're planning to remove containers but wish to keep the data, ensure you've set up Docker volumes **prior to running the container**.

Reviewer: maybe link to volumes

Author: Added.

---
title: First steps with Docker
description: Starting with Docker? Peruse our detailed guide outlining the first steps towards mastering Docker.
---

import { Callout } from 'nextra/components'
import { Steps } from 'nextra/components'

# First steps with Docker

> In this guide, you'll go through the initial steps to install Docker Desktop on Windows 11. By the end, you'll have Docker set up and ready for Memgraph Docker images.

Reviewer: ```suggestion
In this guide, you'll go through the initial steps of installing Docker Desktop on Windows 11. By the end, you'll have Docker set up and ready for Memgraph Docker images.
```

{<h3>Launch Docker Desktop</h3>}

After installation, you can open Docker Desktop by clicking on the newly created icon on your desktop or by searching for it in the Start menu.

{<h3>Accept Docker’s Terms and Conditions</h3>}

Prior to using Docker Desktop, go through the terms and conditions. If you agree, proceed by accepting them.

![Terms and Conditions](/pages/getting-started/first-steps-with-docker/first-steps-with-docker-terms.png)

{<h3>Determine your usage scenario</h3>}

Docker will prompt questions about your intended usage. Answer accordingly to tailor Docker's settings and recommendations to your needs.

{<h3>Post-installation Steps</h3>}

After successful installation, you'll be presented with Docker Desktop’s user interface.

![Docker Desktop UI](/pages/getting-started/first-steps-with-docker/first-steps-with-docker-desktop.png)

Navigate to Docker's settings. Based on your specified usage scenario, verify if Docker is set to start on system boot and adjust this settings.

![Docker Settings](/pages/getting-started/first-steps-with-docker/first-steps-with-docker-settings.png)

<Callout type="info">

> If Docker Desktop is configured not to launch on system startup, you will have to start it manually every time you wish to use it.

Reviewer: ```suggestion
If Docker Desktop is configured not to launch on system startup, you will have to start it manually every time you want to use it.
```

</Callout>

</Steps>

## Verifying Docker Installation

To ensure that Docker has been successfully installed and is running on your system, follow the steps below:

<Steps>
{<h3>Open the Command Line or Terminal</h3>}

- **Windows**: Press `Windows + R`, type `cmd`, and hit Enter.
- **Mac**: Press `Command + Space`, type `terminal`, and hit Enter.
- **Linux**: Depending on the distribution, you can use shortcuts like `Ctrl + Alt + T` or access the terminal from the application menu.

{<h3>Verify Docker Version</h3>}

> Type the following command:

Reviewer: ```suggestion
Run the following command:
```

```
docker --version
```

> This should display the version of Docker installed on your machine. For example:

Reviewer: ```suggestion
It should display the version of Docker installed on your machine. For example:
```

```
Docker version 24.0.6, build ed223bc
```

{<h3>Check Docker Details</h3>}

For more detailed information about the Docker installation, you can use:

```
docker info
```

This command provides a comprehensive overview of your Docker setup, including the number of containers running, paused, and stopped, as well as other system information.

{<h3>Ensure Docker Daemon is Running</h3>}

To check that the Docker daemon (background service) is running, execute:

```
docker system info
```

> If Docker is running correctly, this command will return system-wide information. If not, you may receive an error indicating that Docker is not running.

Reviewer: Maybe an image of what you should expect or C/P from the console? just a suggestion

Author: The output is very informative, but it is to long so I would skip this one.

</Steps>

### Troubleshooting

If you encounter any errors or if Docker does not seem to be running:

- Ensure that the Docker Desktop application (for Mac and Windows) is running.
- On Linux, you might need to start the Docker service manually using a command like `sudo service docker start` or `sudo systemctl start docker`, depending on your Linux distribution.

## Managing Docker Service on Windows using PowerShell

On Windows, Docker typically runs as a service via Docker Desktop. While Docker
Desktop provides a user-friendly interface for managing Docker, there are
instances where you might want to control the Docker service directly from the
command line using PowerShell.

Here's how you can manage the Docker service on Windows using PowerShell:

> ### Open PowerShell as an administrator

Reviewer: I don't think I would have this as a chapter... maybe as a paragraph in the section about Managing DOcker service on windows using powershell

Author: Converted to paragraph.

> - Right-click the Start button and select "Terminal (Admin)" or search

Reviewer: I don't think a bullet is necessary here?

Author: You are right. Removed.

  for "PowerShell" in the start menu, right-click "Windows PowerShell", and
  select "Run as administrator".

### Start the Docker Service

If Docker isn't running and you want to start it, use the following command:

```powershell
net start com.docker.service
```

You should receive a message indicating that the Docker service has started successfully.

### Stop the Docker Service

To stop the Docker service, use the following command:

```powershell
net stop com.docker.service
```

You should receive a message indicating that the Docker service has stopped.

> ### Check the Status of the Docker Service

Reviewer: I think I would move this as a first paragraph, like
1. check if it's running or not
2. start it or stop it depending on what you need

Author: Moved.

```powershell
Get-Service com.docker.service
```

This command will display the status of the Docker service.

## Work with Docker

If you are new to Docker, this guide will help you get a grasp of Docker and
make it easier to accomplish tasks within Memgraph. After installing Docker, all
commands are run from the command-line tool of your preference.

### Download the Memgraph Docker image

You can think about downloading Memgraph Docker images as obtaining a blueprint
for Memgraph, complete with all its core features. This image is your starting
point. From it, you can launch a container, which acts as a live instance of
Memgraph. All your interactions, queries, and transactions with Memgraph occur
within this container. This ensures a consistent and isolated setting for all
your database tasks. From a single image, you can spawn multiple containers.
Each container functions as a distinct Memgraph instance.

Images are downloaded using the `docker pull` command followed by the name of
the Docker image. We encourage you to use Memgraph Platform as it includes
everything you might need while making the best of Memgraph.

> 

Reviewer: I would expand with (just use the normal links, this is how it C/P from the website)

Besides the database, it also includes all the tools you might need to analyze your data, such as command-line interface [mgconsole](https://memgraph.com/docs/getting-started/cli), web interface [Memgraph Lab](https://memgraph.com/docs/data-visualization) and a complete set of algorithms within a [MAGE](https://memgraph.com/docs/advanced-algorithms) library

Author: Done.

To download the latest Memgraph Platform image, run:

```
docker pull memgraph/memgraph-platform
```

All Memgraph Docker images are availble at [Memgraph's Docker
Hub](https://hub.docker.com/u/memgraph). There you will find exact Docker image

> tags you can downalod and use to install Memgraph with. 

Reviewer: ```suggestion
tags you can download and use to install Memgraph with. 
```

### Architecture of Docker container running Memgraph

The picture below shows the architecture of the Memgraph Docker ecosystem.

![](/pages/getting-started/install-memgraph/docker/docker-architecture.png)

Memgraph offers various Docker images, each consisting of different components

- `memgraph-platform` includes Memgraph database, mgconsole, MAGE and Memgraph Lab
- `memgraph-mage` includes Memgraph database, mgconsole and MAGE

> - `memgraph includes` Memgraph database and mgconsole

Reviewer: ```suggestion
- `memgraph` includes Memgraph database and mgconsole
```

Reviewer: maybe link to products?

> We recommend using the `memgraph-platform` image as it includes all the tools

Reviewer: If you're new to Memgraph or you're in a developing stage, we recommend using the `memgraph-platform` image as it includes all the tools you might need to analyze your data. Once you're ready to go to production you can choose to use the `memgraph-mage` Docker image that includes the database and the graph library, or run the database only, using the `memgraph` image.

Author: Updated.

- container name: `--name memgraph`
- set up configuration using environment variables in the case of the
  `memgraph-platform` image, or configuration flags using the `memgraph` or
  `memgraph-mage` image

A `docker run` command can look like this:

```
docker run -p 7687:7687 [-p host_port:container_port] [--name container_name] [configuration] memgraph/image_name
```

#### Publish ports

Ports are published in order to allow services outside the container to be able
to connect to the container. When publishing ports, you need to define two ports
separated by a colon. The left side port stands for the **host port** and the
right side port is the **container port**.

The most common ports published while running Memgraph are:

- `-p 7687:7687` - connection to the database instance (the Bolt protocol uses
  this port by default)
- `-p 3000:3000` - connection to the Memgraph Lab application when running
  Memgraph Platform
- `-p 7444:7444` - connection to fetch log files from Memgraph Lab (only in
  version 2.0 and newer)

For example, if you are running two instances using the `memgraph-platform`
image and you want to connect to both instances using the Memgraph Lab

> in-browser application. You would run the first instance with:

Reviewer: ```suggestion
in-browser application, you would run the first instance with:
```

```
docker run -p 7444:7444 -p 3000:3000 memgraph/memgraph-platform
```

Because port `7444` and `3000` are now taken, you need to change the left side
ports (host ports):

```
docker run -p 7445:7444 -p 3001:3000 memgraph/memgraph-platform
```

To connect to the first instance, you should open Memgraph Lab in your browser
at `localhost:3000`, but the second instance is reachable at `localhost:3001`.

#### Container name

Upon creation, each container gets an ID. You can find out the container ID by
running the `docker ps` command, but you can also name the container using the
`--name` flag. Defining a name can be a handy way to add meaning to a container.
If you specify a name, you can use it when referencing the container within a
Docker network.

#### Set up the configuration

> If you want a certain configuration setting to be applied during a single run,

Reviewer: Link to configuration settings?

Author: Added.

you need to pass the configuration option within the `docker run` command
instead of changing the configuration file.

If you are working with the `memgraph-platform` image, you should pass
configuration options with environment variables.

For example, if you want to limit memory usage for the whole instance to 50 MiB
and set the log level to `TRACE`, pass the configuration like this:

```
docker run -p 7687:7687 -p 3000:3000 -p 7444:7444 -e MEMGRAPH="--memory-limit=50 --log-level=TRACE" memgraph/memgraph-platform
```

When you are working with `memgraph` or `memgraph-mage` images, you should pass
configuration options as arguments.

For example, if you want to limit memory usage for the whole instance to 50 MiB
and set the log level to `TRACE`, pass the configuration argument like this:

```
docker run -p 7687:7687 memgraph/memgraph --memory-limit=50 --log-level=TRACE
```

#### Specify volumes

Specifying a volume syncs the specified directory inside the Docker container as

> a local directory and serves for durability. The `-v` flag is followed by the

Author: Updated.

bind mount will overwrite the content of the Docker container.

For example, if I have a `Data` directory on my `C:` disk, and I want to access
it from inside the container at `/usr/lib/memgraph/data`, I would run Docker
with the following `-v` flag.

```
docker run -p 7687:7687 -p 7444:7444 -p 3000:3000  -v "C:/data":/usr/lib/memgraph/data memgraph/memgraph-platform
```

You can use bind mounts to transfer durability files such as snapshot or WAL
files inside the container to restore data, or CSV files you will use to import
data with `CSV LOAD` clause. 

Bind mounts do not handle data permissions which could cause issues with
permissions. 

### Stop container

<Callout type="info">
Stopping the container isn't the same thing as [deleting it](#containers-and-data-persistence).
</Callout>

Database instances are stopped by stopping the Docker container with the command
`docker stop`. To stop a container you need [to know the container's
ID](#how-to-retrieve-a-docker-container-id) or [name](#container-name).

You can list all the containers you want to stop in one `docker stop` command:

```
docker stop CONTAINER1_ID CONTAINER2_ID CONTAINER3_NAME
```

> ### Start container

Reviewer: I wrote somewhere we could maybe add removing containers as well?

Author: Added.

ID](#how-to-retrieve-a-docker-container-id) or use its [name](#container-name).

Then run the following command if the container ID is `9397623cd87e`.

```console
docker inspect -f '{{range.NetworkSettings.Networks}}{{.IPAddress}}{{end}}' 9397623cd87e
```

### Browse files inside a Docker container

To browse files inside a Docker container, first, you need [to retrieve its
ID](#how-to-retrieve-a-docker-container-id) or use its [name](#container-name).

Then run the following command if the container ID is `9397623cd87e`:

```
docker exec -it 9397623cd87e bash
```

To navigate through the container, use the following commands:

- `ls` - list all the directories and files
- `cd <directory>` - enter a directory
- `cd ..` - leave the directory
- `cat <file>` - list the content of a file

You don't have to write file and directory names in full, once you write enough
letters to have a unique string, press the `<TAB>` key.

### Copy files from and to a Docker container

To copy files from and to the Docker container, first, you need [to retrieve its
ID](#how-to-retrieve-a-docker-container-id) or use its [name](#container-name).

> <Steps>

Reviewer: I'm so sorry... I made a wrong suggestions here...

These can't be really steps because to copying the file from a container or to the container are not really steps 2 and 3, they are both steps 2 depending on what you need... But I also didn't like those number we had before

Do you have an idea how we could improve this? 

Maybe:

First, place yourself in the local directory where you want to copy the file. 

To **copy the file from the container to your current directory** use the following command:

```
docker cp <CONTAINER ID>:<path_in_container> <local_file_name>
```

Be sure to replace the `<CONTAINER ID>` parameter.

The example below will copy the configuration file to the user's Desktop:

```
C:\Users\Vlasta\Desktop>docker cp bb3de2634afe:/etc/memgraph/memgraph.conf memgraph.conf
```

To **copy the file from your current directory to the container** use the following command:

....... the rest of the content

{<h3>Place yourself in the local directory where you want to copy the file.</h3>}

> {<h3>Copy the file from the container to your current directory with the command:</h3>}

Reviewer: ```suggestion
{<h3>Copy the file from the container to your current directory</h3>}

To copy the file from the container to your current directory, use the following command:
```

# First steps with Docker

> In this guide, you'll discover the initial steps to install Docker Desktop on Windows 11. By the end, you'll have Docker set up and ready for Memgraph Docker images.

Reviewer: ```suggestion
In this guide, you'll go through the initial steps to install Docker Desktop on Windows 11. By the end, you'll have Docker set up and ready for Memgraph Docker images.
```

## Installing Docker Desktop on Windows

### Prerequisites 

Ensure your system meets the following prerequisites:

> - **Supported Windows Versions**: 

Reviewer: ```suggestion
- **Supported Windows versions**: 
```

  - Windows 11 64-bit: Home or Pro version 21H2 or higher, or Enterprise or Education version 21H2 or higher.
  - Windows 10 64-bit: Home or Pro 21H2 (build 19044) or higher, or Enterprise or Education 21H2 (build 19044) or higher.

- **WSL**: Ensure you have the WSL 2 feature [activated on Windows](https://learn.microsoft.com/en-us/windows/wsl/install)

> - **Hardware Requirements**: Ensure your machine meets the following criteria:

Reviewer: ```suggestion
- **Hardware requirements**: Ensure your machine meets the following criteria:
```

  - 64-bit processor with Second Level Address Translation (SLAT).
  - At least 4GB system RAM.
  - BIOS-level hardware virtualization must be enabled. 
  

>   Further details can be found [ in the official Docker documenataion](https://docs.docker.com/desktop/install/windows-install/).

Reviewer: ```suggestion
  Further details can be found [in the official Docker documentation](https://docs.docker.com/desktop/install/windows-install/).
```
